File: user-service/src/main/java/com/athena/userservice/model/Invitation.java
package com.athena.userservice.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "invitations")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Invitation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String token;

    @Column(nullable = false)
    private String email;

    @Builder.Default
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "invitation_roles",
            joinColumns = @JoinColumn(name = "invitation_id"),
            inverseJoinColumns = @JoinColumn(name = "role_id"))
    private Set<Role> roles = new HashSet<>();

    @Builder.Default
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "invitation_groups",
            joinColumns = @JoinColumn(name = "invitation_id"),
            inverseJoinColumns = @JoinColumn(name = "group_id"))
    private Set<Group> groups = new HashSet<>();

    @Column(nullable = false)
    private LocalDateTime expiryDate;

    private boolean used;
}

File: user-service/internal/model/invitation.go
package model

import "time"

// Invitation maps to the invitations table.
type Invitation struct {
	ID         int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	Token      string    `gorm:"uniqueIndex;not null" json:"token"`
	Email      string    `gorm:"not null" json:"email"`
	ExpiryDate time.Time `gorm:"column:expiry_date;not null" json:"expiryDate"`
	Used       bool      `gorm:"not null;default:false" json:"used"`
	Roles      []Role    `gorm:"many2many:invitation_roles" json:"roles"`
	Groups     []Group   `gorm:"many2many:invitation_groups" json:"groups"`
}

func (Invitation) TableName() string { return "invitations" }
